src/Microsoft.Extensions.Logging.Abstractions/Internal/FormattedLogValues.cs

```diff
@@ -19,12 +19,23 @@ public class FormattedLogValues : IReadOnlyList<KeyValuePair<string, object>>
         private readonly LogValuesFormatter _formatter;
         private readonly object[] _values;
         private readonly string _originalMessage;
+        private const int MaxCachedFormatters = 1024;
 
         public FormattedLogValues(string format, params object[] values)
         {
             if (values?.Length != 0 && format != null)
             {
-                _formatter = _formatters.GetOrAdd(format, f => new LogValuesFormatter(f));
+                if (_formatters.Count >= MaxCachedFormatters)
```

Comment:
I mean currently the Count check would happen even after the cap is reached but if you were to get a successful hit you could avoid this check.